### PR TITLE
feat(reCAPTCHA): add middleware to verify response

### PR DESCRIPTION
**Pull request by raisedadead**
This PR adds the Google reCAPTCHA API Implementation.

Client side implementation is handled here: 

https://github.com/freeCodeCamp/learn/pull/319
https://github.com/freeCodeCamp/donate/pull/36

### CODE REVIEW

**Comment by raisedadead** [comment]
@bouncey your initial thoughts on this? Do I really need to promisify the call to the Google API?

File: server/boot/donate.js [hunk 1]
```diff
@@ -62,6 +63,40 @@ export default function donateBoot(app, done) {
     });
   }
 
+  function onVerifyreCAPTCHA({ body, connection }, res, next) {
+
+    if (!body || !connection) {
+      return res.status(200).send({ error: 'Captcha validation failed' });
```
Comment (ValeraS) [inline server/boot/donate.js]:
```suggestion
      return res.status(400).send({ error: 'Captcha validation failed' });
```

File: server/boot/donate.js [hunk 2]
```diff
@@ -62,6 +63,40 @@ export default function donateBoot(app, done) {
...
+
+    const verificationURL =
+    'https://www.google.com/recaptcha/api/siteverify?secret=' +
+    keys.recaptcha +
+    '&response=' +
+    captchaResponse +
+    '&remoteip=' +
+    remoteAddress;
+
+    return request(verificationURL, function(error, response, body) {
```
Comment (ValeraS) [inline server/boot/donate.js]: Should it be a post request?

**Review by ValeraS — CHANGES_REQUESTED** [review]
~~You need to rebase your branch for a correct CI build~~

**Comment by raisedadead** [comment]
Thanks @ValeraS ! The CI was erroring out for a random fail in one of the challenges, a restart fixed it. I have also applied your suggestions.

File: server/boot/donate.js [hunk 3]
```diff
@@ -62,6 +63,40 @@ export default function donateBoot(app, done) {
...
+
+    const verificationURL =
+    'https://www.google.com/recaptcha/api/siteverify?secret=' +
+    keys.recaptcha +
+    '&response=' +
+    captchaResponse +
+    '&remoteip=' +
+    remoteAddress;
+
+    return request.post(verificationURL, function(error, response, body) {
```
Comment (ValeraS) [inline server/boot/donate.js]: And params need to place in a request body:
```suggestion
    return request.post({
      url:verificationURL,
      form: {
        secret: keys.recaptcha,
        response: captchaResponse,
        remoteip: remoteAddress
      }
    }, function(error, response, body) {
```

**Review by Bouncey — APPROVED** [review]

**Pull request merged**